hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java

```diff
@@ -113,8 +116,16 @@ private class SstFilteringTask implements BackgroundTask {
     @Override
     public BackgroundTaskResult call() throws Exception {
 
+      Optional<SnapshotCache> snapshotCache = Optional.ofNullable(ozoneManager)
+          .map(OzoneManager::getOmSnapshotManager)
+          .map(OmSnapshotManager::getSnapshotCache);
+      if (!snapshotCache.isPresent()) {
```

Comment:
Does it mean that snapshot is disabled?

Reply:
No SSTFiltering service will not run until snapshot cache becomes available. Just having a safe null check here.

Reply:
If that's the case, I think we should throw an exception if snapshot is enabled and cache is not available otherwise filtering service will not be doing any cleanup.

Reply:
This is just a null check to prevent nullpointer exception

Reply:
Yes, I understand that. But my point is if you expect cache to be non-null then it should be handle properly. If you sure that cache is only null when snapshot is disable then it is fine as I said earlier. But if not, then we should throw an exception when cache is null.

Reply:
We need not throw an exception here. Since it won't be null.

Reply:
As discussed offline, please add a comment that saying that it will be null only when snapshot is disabled or `OmSnapshotManager` is null.